SeaPublicWebsite/Services/RecommendationService.cs

```diff
@@ -1,130 +1,433 @@
-            if (UserDataHelper.HasCavityWalls(userData) && !UserDataHelper.HasInsulatedCavityWalls(userData))
+            return (builtForm, flatLevel);
+        }
+
+        private static string GetConvertedConstructionDate(int? yearBuilt)
+        {
+            return yearBuilt switch
             {
-                userRecommendationKeys.Add(RecommendationKey.InsulateCavityWalls);
-            }
+                <= 1899 => "A",
+                <= 1929 => "B",
+                <= 1949 => "C",
+                <= 1966 => "D",
+                <= 1975 => "E",
+                <= 1982 => "F",
+                <= 1990 => "G",
+                <= 1995 => "H",
+                <= 2002 => "I",
+                <= 2006 => "J",
+                <= 2011 => "K",
+                >= 2012 => "L",
+                _ => "D"
+            };
+        }
 
-            // Not for ground floor or mid floor flat, or if loft is insulated/ flat roof
-            if (userData.FlatType != FlatType.GroundFloor && userData.FlatType != FlatType.MiddleFloor
-                && userData.RoofConstruction != RoofConstruction.Flat
-                && userData.AccessibleLoftSpace == AccessibleLoftSpace.Yes
-                && userData.RoofInsulated != RoofInsulated.Yes
-                && !UserDataHelper.HasRoofInsulation(userData))
+        private static int GetConvertedWallType(WallConstruction? wallConstruction,
+            SolidWallsInsulated? solidWallsInsulated,
+            CavityWallsInsulated? cavityWallsInsulated)
+        {
+            return wallConstruction switch
             {
-                userRecommendationKeys.Add(RecommendationKey.AddLoftInsulation);
-            }
+                WallConstruction.DoNotKnow => (int) WallTypeEnum.DontKnow,
+                WallConstruction.Solid => solidWallsInsulated switch
+                {
+                    SolidWallsInsulated.DoNotKnow => (int) WallTypeEnum.DontKnow,
+                    SolidWallsInsulated.No => (int) WallTypeEnum.SolidWallsWithoutInsulation,
+                    SolidWallsInsulated.Some => (int) WallTypeEnum.SolidWallsWithoutInsulation,
+                    SolidWallsInsulated.All => (int) WallTypeEnum.SolidWallsWithInsulation,
+                    _ => throw new ArgumentOutOfRangeException()
+                },
+                WallConstruction.Cavity => cavityWallsInsulated switch
+                {
+                    CavityWallsInsulated.DoNotKnow => (int) WallTypeEnum.DontKnow,
+                    CavityWallsInsulated.No => (int) WallTypeEnum.CavityWallsWithoutInsulation,
+                    CavityWallsInsulated.Some => (int) WallTypeEnum.CavityWallsWithoutInsulation,
+                    CavityWallsInsulated.All => (int) WallTypeEnum.CavityWallsWithInsulation,
+                    _ => throw new ArgumentOutOfRangeException()
+                },
+                WallConstruction.Mixed =>
+                    //assumption:
```

Comment:
How are we going to make sure that we come back to all of these assumptions? Are they listed on the ticket?